Source/Core/Core/PowerPC/Jit64Common/BlockCache.cpp

```diff
@@ -34,7 +34,7 @@ void JitBlockCache::WriteLinkBlock(const JitBlock::LinkData& source, const JitBl
     else
     {
       Gen::XEmitter emit(location, location + 5);
-      emit.JMP(address, Gen::XEmitter::Jump::Near);
+      emit.JMP(address);
```

Comment:
Wait, I just realized that these things could happen:
- The jump to dispatcher is Near, but the jump to block is Short. That's likely (e.g. jumps to the start of the same block), but it shouldn't cause issues, since this garbage bytes are jumped over. The only issue is that they completely confuse the ASM parser in the debugger... (Granted, can be easily fixed by just adding 3 NOPs, but is 1 Short jump + 3 NOPs any different to 1 Near JMP?)
- The jump to dispatcher is Short, but the jump to block is Near. That would overwrite actual code that is meant to be executed. Is this impossible, or unlikely (meaning that it *can* happen)?

Reply:
> because how much you write here determines where the next block will start
I'm a little confused.
I understand why there is a problem if `JustWriteExit` writes two bytes (which is unlikely but maybe not impossible) but the dispatcher then writes 5.
But in the hypothetical scenario in which `JustWriteExit` has is an optimization (that I was thinking about adding) where a jump to the start of the same block directly writes the address (and thus doesn't go to the dispatcher), then there should be no problem writing two bytes in `JustWriteExit`, right?
(But since this optimization isn't there yet, thus the dispatcher is always called, then 3 NOPs if the JMP is Short have to be there... I guess I could add that optimization now?)

Reply:
No, you have to always write five bytes. `JustWriteExit` is called not only when we want to overwrite the jump with a jump to a block, but also when we want to overwrite the jump with a jump to the dispatcher. See `JitBaseBlockCache::UnlinkBlock`.

Also, please use INT3 instead of NOP. This prevents the CPU from speculating past the end of the block.